packages/@tinacms/toolkit/src/packages/fields/plugins/ImageFieldPlugin.tsx

```diff
@@ -114,16 +112,18 @@ export const ImageField = wrapFieldsWithMeta<InputProps, ImageProps>(
           })
         }}
         onDrop={async ([file]: File[]) => {
+          setIsImgUploading(true)
           const directory = uploadDir(props.form.getState().values)
-
           const [media] = await cms.media.persist([
             {
               directory,
               file,
             },
           ])
-
-          onChange(media)
+          if (media) {
+            await onChange(media)
+            setIsImgUploading(false)
```

Comment:
consider adding a try-catch-finally here or something to deactivate the loading state in the event of an error